openedx/adg/lms/applications/tests/test_views.py

```diff
@@ -0,0 +1,446 @@
+
+    Args:
+        user_for_testing: User object
+        logged_in_client: Client obj with logged in user
+
+    Returns:
+        Boolean
+
+    """
+    ApplicationHub.objects.filter(user=user_for_testing).update(is_prerequisite_courses_passed=True,
+                                                                is_written_application_completed=True,
+                                                                is_application_submitted=True)
+    response = logged_in_client.post(reverse('application_hub'))
+    assert response.status_code == 302
+    assert response.get('Location') == reverse('application_success')
+
+
+# ------- Application Success View tests below -------
+
+@pytest.mark.django_db
+@pytest.fixture()
+def user_with_complete_application(user_for_testing):
+    """
+    Return a user having a complete applicatio to test the success url.
+
+    Args:
+        user_for_testing: User object
+
+    Returns:
+        User object
+
+    """
+
+    ApplicationHub.objects.filter(user=user_for_testing).update(is_prerequisite_courses_passed=True,
+                                                                is_written_application_completed=True,
+                                                                is_application_submitted=True,
+                                                                submission_date=date.today())
+    return user_for_testing
+
+
+@pytest.fixture()
+def get_request_for_application_success_view(request_factory, user_with_complete_application):
+    """
+    Create a get request for the application_success url.
+
+    Args:
+        request_factory: RequestFactory() object.
+        user_with_complete_application: User object
+
+    Returns:
+        HttpRequest obj for application success url.
+
+    """
+    request = request_factory.get(reverse('application_success'))
+    request.user = user_with_complete_application
+    return request
+
+
+def test_get_environment_in_context_for_application_success_view(get_request_for_application_success_view):
+    """
+    Test if the context contains all the necessary pieces.
+
+    Args:
+        get_request_for_application_success_view: HttpRequest obj for application success url
+
+    Returns:
+        Boolean
+
+    """
+    response = ApplicationSuccessView.as_view()(get_request_for_application_success_view)
+    assert 'submission_date' in response.context_data
+
+
+@pytest.mark.django_db
+def test_get_unauthenticated_user_redirects_for_application_success_view():
+    """
+    Test the case where an unauthenticated user is redirected to login page or not.
+
+    Returns:
+        Boolean
+
+    """
+    response = Client().get(reverse('application_success'))
+    assert '/login?next=/application/success' in response.url
+
+
+@pytest.mark.django_db
+def test_get_submission_date_for_application_success_view(user_with_complete_application,
+                                                          get_request_for_application_success_view):
```

Comment:
No need for user_with_complete_application as we have set user in application request object.